source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -117,146 +114,96 @@ void dev_stat_update(mem_stat_update_t type, int16_t size)
     }
 }
 
-//Fix return value allways good fo aligment
 static int heap_alloc_internal_check(int16_t alloc_size)
 {
-    int temp_int, temp_sector = 0, temp_size;
-    temp_size = sizeof(int);
+    int block_count = 0;
```

Comment:
Can we rename this function. It isn't really a "check", apart from as a side effect. It mainly converts requested size to size in words.

     static int convert_allocation_size(int16_t requested_bytes)

or something like that?

And make the return variable "words" or something - block count sounds to me like "number of blocks".